### PR TITLE
feat(FocusRing): mvp for FocusRing component

### PR DESCRIPTION
**Pull request by ctoddy**
## Description

This is the mvp of the focusRing component. It will require more work on the props and styling but this is a good stopping point to review. 

https://github.com/user-attachments/assets/e144170c-f881-4011-b9c5-ca24c58da9e8

## Changes

Created focusRing and supporting setFocusRing utility.

## Testing

Use Focus Ring on top of buttons or any other component that utilizes focus. See the storybook docs for this story on how to set up for use.

### CODE REVIEW

File: packages/solid/components/FocusRing/FocusRing.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import styles from './FocusRing.styles.js';
+
+const FocusRing: Component<FocusRingProps> = (props: FocusRingProps) => {
+  return (
+    <View
+      {...props}
+      ref={props.ref}
+      radius={props.radius}
+      width={props.width}
+      height={props.height}
```
Comment (anthony9187) [inline packages/solid/components/FocusRing/FocusRing.tsx:30]: we can remove these as `{...props}` will handle passing them along

File: packages/solid/components/FocusRing/FocusRing.types.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import type { UIComponentProps } from 'types/interfaces.js';
+//import type { Signal } from 'solid-js';
+import type { ComponentStyleConfig, NodeStyleSet, Tone } from 'types/types.js';
+import type { NodeStyles } from '@lightningtv/solid';
+
+export interface FocusRingProps extends UIComponentProps {
+  /**
+   * placeholder for FocusRingProps
+   */
+}
```
Comment (anthony9187) [inline packages/solid/components/FocusRing/FocusRing.types.ts:27]: we can leave this for now, but if we wind up without any FocusRing specific props we should remove it

File: packages/solid/components/FocusRing/FocusRing.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * Add a focus ring to tile or other items to provide a nice highlight
+ * and shadow effect
+ *
+ * ```js
+ * import { View } from '@lightningtv/solid';
+ * import { useFocusManager } from '@lightningtv/solid/primitives';
+ * import { FocusRing } from '@lightningtv/solid-ui';
+ * import { setFocusRing } from '@lightningtv/solid-ui/utils';
+ *
+ * const App = () => {
```
Comment (anthony9187) [inline packages/solid/components/FocusRing/FocusRing.stories.tsx:36]: is the plan to restore this in the future?

**Comment by anthony9187** [comment]
Looking great! Can we also try and add some information around leveraging the FocusRing in an application to the docs story? I can't decide if it makes more sense for the setFocusRing util to live within the component package itself 🤔

File: packages/solid/components/FocusRing/FocusRing.stories.tsx (hunk 3 above)
Comment (ctoddy) [reply packages/solid/components/FocusRing/FocusRing.stories.tsx:36]: No this is the "How to use docs." You insert the docs in the stories file now with the new use of autodocs instead of the mdx.  I remember we wanted to add this information for the focusRing since it has a specific set up so this is just an example of how to use it. 
Screenshot 2024-08-20 at 8 35 32 AM

**Comment by ctoddy** [comment]
Yep, that's what I did in the stories with the autodocs. I'll check how other component-specific utils are set up in other repos to see if the setFocusRing should move within its component package.

**Comment by anthony9187** [comment]
> Yep, that's what I did in the stories with the autodocs. I'll check how other component-specific utils are set up in other repos to see if the setFocusRing should move within its component package.

Yup I'm seeing the implementation example now, that's great. It'd be helpful to also include an explanation of what's going on and why - setups like this can be hard to parse without context and I think it'd be best to try and provide it where we can

**Comment by ctoddy** [comment]
Agreed and will do 😃

File: packages/solid/components/FocusRing/setFocusRing.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createEffect, on } from 'solid-js';
+import { activeElement } from '@lightningtv/solid';
+
+export const setFocusRing = focusRingRef =>
```
Comment (anthony9187) [inline packages/solid/components/FocusRing/setFocusRing.ts:21]: we should add a type for this `focusRingRef` - my vote's for `SolidNode` but @chiefcll might have a better recommendation for a ref vs a node

File: packages/solid/components/FocusRing/FocusRing.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+ * - `setFocusRing`: This function in the FocusRing package is used to set the FocusRing reference and position the component on the current `activeElement`.
+ * - `FocusRing` : The FocusRing itself.
+ *
+ * ```js
+ * import { onMount } from 'solid-js';
+ * import { useFocusManager } from '@lightningtv/solid/primitives';
+ * import { FocusRing, setFocusRing } from '@lightningtv/solid-ui';
+ * ```
+ * #### 2. Create a ref:
+ * Declare a ref variable that will hold a reference to the `FocusRing` component. For example `let focusRef;`
```
Comment (anthony9187) [inline packages/solid/components/FocusRing/FocusRing.stories.tsx:40]:
```suggestion
 * Declare an empty variable that will be used to store a reference to the `FocusRing` component. For example `let focusRef;`
```

File: packages/solid/components/FocusRing/FocusRing.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+type Story = StoryObj<typeof FocusRing>;
+
+/**
+ * Fancy ring to focus on Items.
+ *
+ * ## Usage
+ * These steps set up a FocusRing component and associates it with a reference. This reference is then used by the setFocusRing function and the focus manager to set the focus ring to the current activeElement.
+ * #### 1. Necessary imports:
+ *
+ * - `useFocusManager`: This primitive is used to manage focus within the application. See more on `useFocusManager` [here](https://lightning-tv.github.io/solid/#/primitives/useFocusManager?id=usefocusmanager-for-key-handling).
```
Comment (anthony9187) [inline packages/solid/components/FocusRing/FocusRing.stories.tsx:29]:
```suggestion
 * - `useFocusManager`: an Effect is used to manage focus within the application. See more on `useFocusManager` [here](https://lightning-tv.github.io/solid/#/primitives/useFocusManager?id=usefocusmanager-for-key-handling).
```

**Review by anthony9187 — APPROVED** [review]
